src/lib.rs

```diff
@@ -387,6 +392,68 @@ impl Node {
     }
 }
 
+impl Parser {
+    pub fn new(mime_type: &str) -> Result<Self, i32> {
+        let mime = CString::new(mime_type).map_err(|_| -1)?;
+        unsafe {
+            let parser = librdf_new_parser(
+                unwrap!(WORLD.lock()).as_ptr(),
+                ptr::null(),
+                mime.as_ptr() as *mut _,
+                ptr::null_mut(),
+            );
+            if parser.is_null() {
+                return Err(-1);
+            }
+            Ok(Parser::from_ptr(parser))
+        }
+    }
+
+    pub fn parse_string(
+        parser: Parser,
+        syntax: &str,
+        base_uri: Uri,
+        model: &Model,
+    ) -> Result<(), i32> {
+        let converted_syntax = CString::new(syntax).map_err(|_| -1)?;
+        unsafe {
+            let res = librdf_parser_parse_string_into_model(
+                parser.as_ptr(),
+                converted_syntax.as_ptr() as *mut _,
+                base_uri.as_ptr(),
+                model.as_ptr(),
+            );
+            if res > 0 {
+                return Err(-1);
+            }
+            Ok(())
+        }
+    }
+
+    pub fn parse_from_file(
+        parser: Parser,
+        file_handle: *mut FILE,
+        base_uri: Uri,
+        model: &Model,
+    ) -> Result<(), i32> {
+        unsafe {
+            let res = librdf_parser_parse_file_handle_into_model(
+                parser.as_ptr(),
+                file_handle as *mut redland_sys::_IO_FILE,
```

Comment:
Would type inference work here? I.e.:
```suggestion
                file_handle as *mut _,
```